src/Vehicle/InitialConnectStateMachine.cc

```diff
@@ -353,6 +360,10 @@ void InitialConnectStateMachine::_stateRequestRallyPoints(StateMachine* stateMac
     Vehicle*                    vehicle         = connectMachine->_vehicle;
     SharedLinkInterfacePtr      sharedLink      = vehicle->vehicleLinkManager()->primaryLink().lock();
 
+    if (vehicle->_geoFenceManager->supported() && vehicle->_geoFenceManager != nullptr)
```

Comment:
You could just unconditionally disconnect (_geoFenceManager is always initialized).
```suggestion
    if (vehicle->_geoFenceManager != nullptr && vehicle->_geoFenceManager->supported())
```

Reply:
Right. I removed the paranoia checks and just disconnect unconditionally. As you pointed out my paranoia check was not even correct... Thanks for reviewing the code!